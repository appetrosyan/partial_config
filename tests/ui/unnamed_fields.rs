use partial_config::HasPartial;

#[derive(Debug, HasPartial)]
pub struct Conf(Option<u64>);

#[derive(HasPartial)]
pub struct Config;

fn main() {}
