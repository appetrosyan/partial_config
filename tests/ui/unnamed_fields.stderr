error: Unnamed fields can't be named in configuration layers.

         = help: If the field is unnamed, I cannot find a consistent way of naming them in configuration layers, because they muse be human facing. You are probably applying this derive macro to a tuple structure, which is not a sensible input.

 --> tests/ui/unnamed_fields.rs:4:16
  |
4 | pub struct Conf(Option<u64>);
  |                ^^^^^^^^^^^^^

error: Unit fields cannot be named.

         = help: If the field is unnamed, I cannot find a consistent way of naming them in configuration layers. Add a dummy field with e.g. `PhantomData` to silence this error.

 --> tests/ui/unnamed_fields.rs:6:10
  |
6 | #[derive(HasPartial)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `HasPartial` (in Nightly builds, run with -Z macro-backtrace for more info)
